### PR TITLE
Fast path for heatmap

### PR DESCRIPTION
**Pull request by SimonDanisch**
This brings back the fast path for heatmaps, if there are no transformations and the heatmap is on a regular grid.

```julia
using WGLMakie, BenchmarkTools
@btime(
    begin
        heatmap!(s, (-1, 1), (-1, 1), data)
        display(s)
    end,
    setup = (data=rand(4000, 4000); s = Scene())
)
# PR, fix
212.649 ms (4410 allocations: 38.34 MiB)
# Master
295.968 ms (4341 allocations: 541.18 MiB)
```
Before, WGLMakie would create an `NxM` mesh, which is quite big (as can be seen in the allocations).
This was slower and lots of data to transfer.

This PR needed to adjust the conversion pipeline to not convert any `1..x` to `[1...x]`, which brings it's own set of problems that still needs cleaning up for this PR to pass tests.

### CODE REVIEW

**Comment by SimonDanisch** [comment]
The WGLMakie failure seems to be an harmless artifact of slightly different interpolations from mesh -> image rendering for a regular heatmap.
@ffreyer, what do you think, I'd just update the reference Image and have this merged.

**Comment by ffreyer** [comment]
Yea, they look harmless and sometimes even a bit better

File: GLMakie/src/glshaders/mesh.jl [hunk 1]
```diff
@@ -50,7 +50,7 @@ function draw_mesh(screen, data::Dict)
         color_norm = nothing
         fetch_pixel = false
         texturecoordinates = Vec2f(0) => GLBuffer
-        uv_transform = Vec4f(1, 1, 0, 0)
+        uv_transform = Mat{2,3,Float32}(1, 0, 0, -1, 0, 1)
```
Comment (ffreyer) [inline GLMakie/src/glshaders/mesh.jl:53]: Maybe we should fix this in surface as well?

File: WGLMakie/src/imagelike.jl [hunk 2]
```diff
@@ -163,26 +163,22 @@ function limits_to_uvmesh(plot, f32c)
...
 
-    # TODO, this branch is only hit by Image, but not for Heatmap with stepranges
-    # because convert_arguments converts x/y to Vector{Float32}
-    if px[] isa Makie.Interval && py[] isa Makie.Interval && Makie.is_identity_transform(t)
+    if px[] isa Makie.EndPoints && py[] isa Makie.EndPoints && Makie.is_identity_transform(t)
         rect = lift(plot, px, py) do x, y
-            xmin, xmax = Makie.endpoints(x)
-            ymin, ymax = Makie.endpoints(y)
+            xmin, xmax = extrema(x)
+            ymin, ymax = extrema(y)
```
Comment (ffreyer) [inline WGLMakie/src/imagelike.jl:169]: Not sure if this is something to do here or in #3989, but a lot of these extrema probably need to be replaced with splatting or `x[1], x[2]` if we want to support reversed images, heatmaps etc

Comment (SimonDanisch) [reply WGLMakie/src/imagelike.jl:169]: Yeah, I wouldn't want to do it here, and sticking with extrema makes it easier to find once we get to it.

**Pull request merged**